From: Chris Kelly <c-kelly@ti.com>
Date: Fri, 21 Jan 2011 21:25:41 +0000 (-0600)
Subject: OMAP4: Vibrator: Use LP PLL for vibrator
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=752addc67fbe57284523f7770a605cf5d0dbdc1f;hp=c96e270a05eadda558d3ce1b6529d85f95b3217e

OMAP4: Vibrator: Use LP PLL for vibrator

        Audio and vibrator are controlled by twl6040. To maintain consistency
        with audio, vibrator should use LP PLL as well. Long term solution
        should be to abstract Phoenix low-level configuration.

This reverts commit d92ad9564343d3e5ae6e0005e26c6596453c849 which is a revert
of the original commit to use LP PLL for vibrator.

Change-Id: Ib876b8a31ac1751c22c0e2d035ec66115d9c995b
Signed-off-by: Chris Kelly <c-kelly@ti.com>
---

Index: android-2.6.35/drivers/misc/twl6040-vib.c
===================================================================
--- android-2.6.35.orig/drivers/misc/twl6040-vib.c	2011-01-24 11:50:00.000000000 +0530
+++ android-2.6.35/drivers/misc/twl6040-vib.c	2011-01-25 11:51:46.707842748 +0530
@@ -54,18 +54,21 @@
 		 *	 ensure syncronization between audio and vibra
 		 *	 components.
 		 */
-		hppllctl = TWL6040_MCLK_38400KHZ | TWL6040_HPLLSQRENA |
-			   TWL6040_HPLLBP | TWL6040_HPLLENA;
-		hppllctl &= ~TWL6040_HPLLSQRBP & ~TWL6040_HPLLRST;
-		twl_i2c_write_u8(TWL4030_MODULE_AUDIO_VOICE,
-				hppllctl, TWL6040_REG_HPPLLCTL);
-		lppllctl = TWL6040_HPLLSEL | TWL6040_LPLLENA;
-		twl_i2c_write_u8(TWL4030_MODULE_AUDIO_VOICE,
-				lppllctl, TWL6040_REG_LPPLLCTL);
-		udelay(100);
-		lppllctl &= ~TWL6040_LPLLENA;
+		lppllctl = TWL6040_LPLLENA;
 		twl_i2c_write_u8(TWL4030_MODULE_AUDIO_VOICE,
-				lppllctl, TWL6040_REG_LPPLLCTL);
+				 lppllctl, TWL6040_REG_LPPLLCTL);
+		mdelay(5);
+		lppllctl &= ~TWL6040_HPLLSEL;
+		twl_i2c_write_u8(TWL4030_MODULE_AUDIO_VOICE,
+				 lppllctl, TWL6040_REG_LPPLLCTL);
+		twl_i2c_read_u8(TWL4030_MODULE_AUDIO_VOICE,
+				&hppllctl, TWL6040_REG_HPPLLCTL);
+		hppllctl &= ~TWL6040_HPLLENA;
+		twl_i2c_write_u8(TWL4030_MODULE_AUDIO_VOICE,
+				 hppllctl, TWL6040_REG_HPPLLCTL);
+		lppllctl &= ~TWL6040_LPLLFIN;
+		twl_i2c_write_u8(TWL4030_MODULE_AUDIO_VOICE,
+				 lppllctl, TWL6040_REG_LPPLLCTL);
 
 		twl_i2c_read_u8(TWL4030_MODULE_AUDIO_VOICE,
 				&reg, TWL6040_REG_VIBCTLL);
